zebra-chain/src/fmt.rs

```diff
@@ -13,6 +14,21 @@ where
     }
 }
 
+impl<T> ops::Deref for DisplayToDebug<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T> From<T> for DisplayToDebug<T> {
+    fn from(t: T) -> Self {
+        Self(t)
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
```suggestion
/// Wrapper to override `Debug` to display a shorter summary of the type.
///
/// For `Vec<T>`s, it only displays the item type name and the length.
#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Not necessary, I just thought it might be useful :)

Reply:
Thanks! Applied as part of 76c0b074242d717dae56aaf275862fc4c51f4da8